packages/core/admin/admin/src/content-manager/pages/ListView/components/ViewSettingsMenu/index.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+
+import { Flex, IconButton, Popover } from '@strapi/design-system';
+import { CheckPermissions, LinkButton } from '@strapi/helper-plugin';
+import { Cog, Layer } from '@strapi/icons';
+import PropTypes from 'prop-types';
+import { useIntl } from 'react-intl';
+import { useSelector } from 'react-redux';
+
+import { selectAdminPermissions } from '../../../../../pages/App/selectors';
+import { FieldPicker } from '../FieldPicker';
+
+export const ViewSettingsMenu = ({ slug, layout }) => {
+  const [isVisible, setIsVisible] = React.useState(false);
+  const cogButtonRef = React.useRef();
+  const permissions = useSelector(selectAdminPermissions);
+  const { formatMessage } = useIntl();
+
+  const handleToggle = () => {
+    setIsVisible((prev) => !prev);
+  };
+
+  return (
+    <Flex justifyContent="flex-end" height="100%">
+      <IconButton
+        icon={<Cog />}
+        label={formatMessage({
+          id: 'components.ViewSettings.tooltip',
+          defaultMessage: 'View Settings',
+        })}
+        ref={cogButtonRef}
+        onClick={handleToggle}
+      />
+      {isVisible && (
+        <Popover
+          placement="bottom-end"
+          source={cogButtonRef}
+          isVisible={isVisible}
+          onDismiss={handleToggle}
+          spacing={4}
+        >
+          <Flex direction="column" gap={3} padding={2}>
+            <CheckPermissions
```

Comment:
Shouldn't this surround the whole component? I'd find it confusing if the user could open the menu, but it doesn't contain anything.

Reply:
The CheckPermissions function only surrounds the "Configure the view" button, not the checkboxes or the Reset button. Therefore, if you don't have permissions, only the "Configure the view" button will be hidden.

Reply:
Makes sense.